### ISSUE TITLE
Use Runtime Keys by WinterCG

### ISSUE THREAD

**Issue by yusukebe** [honojs/hono#844]
`c.runtime` returns which runtime is currently running on. The current keys are below:

```ts
type Runtime = 'node' | 'deno' | 'bun' | 'cloudflare' | 'fastly' | 'vercel' | 'lagon' | 'other'
```

Recently the WinterCG is [defining the "Runtime Keys"](https://github.com/wintercg/runtime-keys/pull/2). We can see the draft in the below link:

* <https://runtime-keys.proposal.wintercg.org>

According to this document, there are other names for the keys than the ones we have defined. For example, we defined the Cloudflare Workers as `cloudflare`, but WinterCG named it `workerd`.

Since Hono should be WinterCG compliant, we should use these keys (`workerd` etc.). The only thing is that Fastly has not joined WinterCG, so that should stay as it is.

This change will be a Breaking Change, so let's add it to the [next branch](https://github.com/honojs/hono/tree/next).

**Comment by yusukebe** [comment]
Hi @yujong-lee !

Are you interested in this matter? I would like to hear your opinion.

**Comment by yujong-lee** [comment]
@yusukebe 
Yes, I think we should use the same key as WinterCG. Free from naming things!

It seems like a change as below will be enough. What do you think?
```ts
type Runtime = 'node' | 'deno' | 'bun' | 'workerd' | 'fastly' | 'edge-light' | 'lagon' | 'other'
```

**Comment by yusukebe** [comment]
> It seems like a change as below will be enough. What do you think?

Looks good! Could you create the PR? If you do so, make the PR into [the next branch](https://github.com/honojs/hono/tree/next).

**Comment by yusukebe** [comment]
This is fixed by #850
